client/device/cloudftd/cloudftdonboarding/create_test.go

```diff
@@ -0,0 +1,425 @@
+package cloudftdonboarding_test
+
+import (
+	"context"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/device/cloudftd/cloudftdonboarding"
+	internalHttp "github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/http"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/url"
+	"github.com/jarcoal/httpmock"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"testing"
+	"time"
+)
+
+func TestCloudFtdOnboardingCreate(t *testing.T) {
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	testCases := []struct {
+		testName   string
+		input      cloudftdonboarding.CreateInput
+		setupFunc  func()
+		assertFunc func(output *cloudftdonboarding.CreateOutput, err error, t *testing.T)
+	}{
+		{
+			testName: "successful ftd onboarding",
+			input:    cloudftdonboarding.NewCreateInput(ftdUid),
+			setupFunc: func() {
+				ReadFmcIsSuccessful(true)
+				ReadFmcDomainInfoIsSuccessful(true)
+				ReadApiTokenInfoIsSuccessful(true)
+				CreateSystemApiTokenIsSuccessful(true)
+				ReadFtdMetadataIsSuccessful(true)
+				CreateFmcDeviceRecordIsSuccessful(true)
+				ReadTaskStatusIsSuccessful(true)
+				ReadFtdSpecificDeviceIsSuccessful(true)
+				TriggerRegisterFmcStateMachineSuccess(true)
+				TriggerRegisterFmcStateMachineEndsInDone(true)
+			},
+			assertFunc: func(output *cloudftdonboarding.CreateOutput, err error, t *testing.T) {
+				assert.Nil(t, err)
+				assert.NotNil(t, output)
```

Comment:
What I don't like about these kinds of test cases is that we're not checking the output against an expected value.

Reply:
added some checks

Reply:
Thanks!